### ISSUE TITLE
cranelift-wasm: translate wasm spec test suite to cranelift IR and run it through the verifier

### ISSUE THREAD

**Issue by fitzgen** [CraneStation/cranelift#1096]
Actually running the spec test suite requires an embedder, but even if we just translate it to cranelift ir and verify what we produce, we should get useful tests out of it. This would have helped me catch some bugs with my multi-value wasm PR, that I only later ended up finding out about when working on the wasmtime integration.

**Comment by bnjbvr** [comment]
Agreed, even compiling the test cases would at least catch issues during compilation. Adding wat tests from the wasm spec test suite into the wasm test directory is one possibility. Is this what you are proposing we should do here?

**Comment by fitzgen** [comment]
> Adding wat tests from the wasm spec test suite into the wasm test directory is one possibility. Is this what you are proposing we should do here?

Yeah, as a git submodule, so it is easy to keep up to date.
